src/Core/IdentityServer/ResourceOwnerPasswordValidator.cs

```diff
@@ -135,5 +134,33 @@ protected async override Task<(User, bool)> ValidateContextAsync(ResourceOwnerPa
         {
             context.Result = new GrantValidationResult(TokenRequestErrors.InvalidGrant, customResponse: customResponse);
         }
+
+        private bool AuthEmailHeaderIsValid(ResourceOwnerPasswordValidationContext context)
+        {
+            if (!_currentContext.HttpContext.Request.Headers.ContainsKey("Auth-Email"))
+            {
+                return false;
+            }
+            else
+            {
+                try
+                {
+                    var authEmailHeader = _currentContext.HttpContext.Request.Headers["Auth-Email"];
+                    var authEmailDecoded = CoreHelpers.Base64UrlDecodeString(authEmailHeader);
+
+                    if (authEmailDecoded != context.UserName)
+                    {
+                        return false;
```

Comment:
This is okay for this fix, however the preference for a simple boolean return would be something like this for simplification (without the `if` statement):

```c#
return authEmailDecoded == context.UserName;
```

Reply:
I did wonder about this but decided against it - will know for next time! Thanks